doc/media/openage/texture_format_spec.md

```diff
@@ -0,0 +1,208 @@
+# Texture Format Specification
+
+**Format Version:** 1
+
+The openage texture format is a plaintext configuration file format for defining 2D
+textures and contained subtextures. It tells the openage renderer which image resource
+it has to load and how the resource s partioned into subtextures.
+
+All attributes start with a defined keyword followed by parameter values. Some
+parameters may have default values and are optional. The preferred file extension is
+`.texture`.
+
+A `.texture` file should be placed next to the image file it references and have the same
+filename stem.
+
+
+## Quick Reference
+
+```
+# This is a texture configuration file
+# comments start with # and are ignored
+
+# file version
+version 1
+
+# image file reference, relative to this file's location
+imagefile <filename>
+
+# Image size
+size <width> <height>
+
+# Pixel format representation
+pxformat <format> cbit=<token>
+
+# Defines a subtexture inside the image resource
+# Subtextures have x and y coordinates, width and height
+# as well as anchor points.
+subtex <id> <xpos> <ypos> <xsize> <ysize> <xhotspot> <yhotspot>
+```
+
+
+## Data Type Formatting
+
+Type     | Example | Description
+---------|---------|---------
+int      | `5`     | Signed Integer
+string   | `"bla"` | String of characters enclosed by `"`
+token    | `off`   | Alphanumeric predefined keyword
+
+
+## Attributes
+
+### `version`
+
+Version of the texture format. Increments every time the syntax
+or keywords of the format change.
+
+Parameter  | Type   | Optional | Default value
+-----------|--------|----------|--------------
+version_no | int    | No       | -
+
+**version_no**<br>
+Version number of the format.
+
+#### Example
+
+```
+version 1
+```
+
+
+### `imagefile`
+
+Tells the renderer which image resource it has to load.
+There has to be exactly one `imagefile` defined.
+
+Parameter | Type   | Optional | Default value
+----------|--------|----------|--------------
+filename  | string | No       | -
+
+**filename**<br>
+Path to the image resource on the filesystem. The different methods of
+resource referencing are explained in the [file referencing](file_referencing.md)
+docs.
+
+
+#### Example
+
+```
+imagefile "idle.png"
+```
+
+
+### `size`
+
+Size of the image loaded from the file.
+
+Parameter | Type   | Optional | Default value
+----------|--------|----------|--------------
+width     | int    | No       | -
+height    | int    | No       | -
+
+**width**<br>
+Width of the image in pixels.
+
+**height**<br>
+Height of the image in pixels.
+
+
+#### Example
+
+```
+size 800 600
+```
+
+
+### `pxformat`
+
+Pixel format used for pixels in the image.
+
+Parameter | Type   | Optional | Default value
+----------|--------|----------|--------------
+format    | token  | No       | -
+cbit      | token  | Yes      | `false`
+
+**format**<br>
+Pixel format of the image.
+
+Format    | Description
+----------|------------
+`rgba8`   | 32 bits per pixel, RGBA colours
```

Comment:
In this case it could be, yes. However, there could be more funky custom openage formats in the future (blend files, etc.) that need a special interpretation, so I thought it would be better to cover this here.